src/lib/check-medic-conf-dependency-version.js

```diff
@@ -1,70 +1,36 @@
 /**
  * Attempt to check that the version of medic-conf that a project relies on
  * matches the version of medic-conf being used to configure that project.
- *
- * Currently this check is only performed as part of app-settings compilation;
- * if required in more places then we might move the check to the main CLI
- * script.
  */
 
-const error = require('./log').error;
 const fs = require('./sync-fs');
-const readline = require('readline-sync');
 const semver = require('semver');
 const { warn } = require('./log');
 
-module.exports = projectDir => {
-  const myVersion = require('../../package.json').version;
+const runningVersion = require('../../package.json').version;
 
-  const theirPackageVersion = readRequestedVersion(projectDir);
-  if(!theirPackageVersion) {
+module.exports = projectDir => {
+  const projectVersion = readRequestedVersion(projectDir);
+  if(!projectVersion) {
     warn('Project has no dependency on medic-conf.');
     return;
   }
 
-  if(!semver.satisfies(myVersion, theirPackageVersion)) {
-    throw new Error(`medic-conf version ${myVersion} does not match the project's required version: ${theirPackageVersion}`);
+  const majorRunningVersion = semver.major(runningVersion);
+  if(semver.satisfies(projectVersion, `<${majorRunningVersion}.x || >${runningVersion}`)) {
+    throw new Error(`medic-conf version ${runningVersion} does not match the project's required version: ${projectVersion}. To ignore this error, use --skip-dependency-check.`);
```

Comment:
Can we workshop this error message a little? It's great that it gives you an action to perform to keep moving, but because it doesn't make the app's reasoning clear for blocking progress, and users being users, I worry they will just read that and instantly reach for the flag (#YOLO) so they can move past the annoyance and keep going.

Try to mix in the additional points of:
 - this means your project may not compile or work as you'd expect, which is bad
 - you can solve the problem by updating your medic-conf version (which is almost certainly the correct approach right, compared to the yolo flag)

Reply:
@SCdF ok, I'll go with 
`medic-conf version ${myVersion} does not match the project's required version: ${theirPackageVersion}. So the project may not compile or work as expected. To resolve this, update your medic-conf or [we do not advise this] work around the error using the --skip-dependency-check flag`

Reply:
I realised when working through how I'd make this clearer there are actually two solutions that the user needs to consider: 
 - if the project is on 3.2.1 and they are on 3.2.0 they need to update their local medic-conf
 - if the project is on 3.2.1 and they are on 4.0.0 they need to either downgrade their local medic-conf, or correctly update the project to be 4.0 compatible

Here is my attempt:

```
Your medic-conf version is incompatible with the project's medic-conf version:
Yours:  ${myVersion}
Theirs: ${theirPackageVersion}

Continuing without updating could cause this project to not compile or work as expected.

Update your local medic-conf with:
 npm i -g medic-conf@${theirPackageVersion}
and try again, or update your project to your current version, or ignore this warning with --skip-dependency-check
```

This is sort of following the git modal: big intentionally clear block of text that gives you the best option as something you can copy paste.

I am not happy with "or update your project to your current version" as it doesn't really make it clear how, though. Thoughts?

Reply:
`if the project is on 3.2.1 and they are on 3.2.0 they need to update their local medic-conf
if the project is on 3.2.1 and they are on 4.0.0 they need to either downgrade their local medic-conf, or correctly update the project to be 4.0 compatible`

I have made the message dynamic to note this to the user.

`Your medic-conf version is incompatible with the project's medic-conf version:
Yours:  ${myVersion}
Theirs: ${theirPackageVersion}

Continuing without updating could cause this project to not compile or work as expected.

Update your local medic-conf with:
 npm i -g medic-conf@${theirPackageVersion}
and try again, or update the project to your current version, or ignore this warning with --skip-dependency-check`

This message captures the points so I have used it almost word-for-word.

Reply:
`
Your medic-conf version is incompatible with the project's medic-conf version:
Yours:  ${myVersion}
Theirs: ${theirPackageVersion}

Continuing without updating could cause this project to not compile or work as expected.

Update your local medic-conf with:
 npm i -g medic-conf@${theirPackageVersion}
and try again, or update the project to your current version, or ignore this warning with --skip-dependency-check
`

This message captures the points so I have used it almost word-for-word.